service/src/main/java/org/mskcc/cmo/metadb/service/util/DataTransformer.java

```diff
@@ -0,0 +1,111 @@
+        researchSampleMapping.put("cmoSampleClass", "sampleType");
+        researchSampleMapping.put("specimenType", "sampleClass");
+        researchSampleMapping.put("oncoTreeCode", "oncotreeCode");
+        researchSampleMapping.put("requestId", "igoRequestId");
+        researchSampleMapping.put("recipe", "genePanel");
+        researchSampleMapping.put("igoId", "primaryId");
+        return researchSampleMapping;
+    }
+    
+    Map<String, String> setUpMetastasisValueMapping() {
+        Map<String, String> metastasisValueMapping = new HashMap<>();
+        metastasisValueMapping.put("0", "Primary");
+        metastasisValueMapping.put("1", "Metastasis");
+        metastasisValueMapping.put("2", "Local Recurrence");
+        metastasisValueMapping.put("127", "Unknown");
+        return metastasisValueMapping;
+    }
+
+    private final ObjectMapper mapper = new ObjectMapper();
+
+    public String transformRequestMetadata(String requestJson) throws JsonProcessingException {
+        return updateMapKeyValueBasedOnMapping(requestJson, setUpResearchRequestMapping());
+    }
+
+    public String transformResearchSampleMetadata(String sampleJson) throws JsonProcessingException {
+        return updateMapKeyValueBasedOnMapping(sampleJson, setUpResearchSampleMapping());
+    }
+
+    /**
+     * transformClinicalSampleMetadata
+     * @param sampleJson
+     * @return
+     * @throws JsonProcessingException
+     */
+    public String transformClinicalSampleMetadata(String sampleJson)
+            throws JsonProcessingException {
+        String transformedSampleJson = updateMapKeyValueBasedOnMapping(
+                sampleJson, setUpClinicalSampleMapping());
+        
+        Map<String, Object> transformedSampleMetadataMap = mapper.readValue(transformedSampleJson, Map.class);
+        String primaryId = mapper.writeValueAsString(transformedSampleMetadataMap.get("primaryId"));
+
+        String sampleType = setUpMetastasisValueMapping().get(transformedSampleMetadataMap.get("sampleType"));
+        transformedSampleMetadataMap.replace("sampleType", sampleType);
+
+        String sampleClass = primaryId.matches("[ACCESS REGEX PATTERN]") ? "cfDNA" : "Tumor";
+        transformedSampleMetadataMap.replace("sampleClass", sampleClass);
+
+        String sex = transformedSampleMetadataMap.get("gender").toString().equals("0") ? "Male" : "Female";
+        transformedSampleMetadataMap.replace("sex", sex);
+
+        String tumorOrNormal = primaryId.matches("DMP NORMAL SAMPLE SUFFIX PATTERN") ? "Normal" : "Tumor";
+        transformedSampleMetadataMap.replace("tumorOrNormal", tumorOrNormal);
+
+        return transformedSampleJson;
+    }
+
+    /**
+     * updateMapKeyValueBasedOnMapping
+     * @param unParsedJson
+     * @param universalMapping
+     * @return
+     * @throws JsonProcessingException
+     */
+    public String updateMapKeyValueBasedOnMapping(String unParsedJson,
```

Comment:
before pursing this strategy it would be helpful to have unit tests confirming that the transformed map/json contains the expected changes